### ISSUE TITLE
Display beatmap info inside a multiplayer room

### ISSUE THREAD

**Issue by Theighlin** [ppy/osu#11306]
**Describe the new feature:** Currently you have to click on the beatmap name and open its page to see info like lenght, star rating etc... It should be displayed in the room screen like in stable.

**Proposal designs of the feature:** it could be placed in the upper-right corner above the track for now.

**Comment by peppy** [comment]
@arflyte is this something that has been given any consideration? Is there an extended beatmap panel or some kind of display which could be used in places like this to give more context? Or should there be a mutliplayer room specific design showing this kind of information?

**Comment by arflyte** [comment]
Haven't given this any thought yet. I'll design the extra info.

**Comment by smallketchup82** [comment]
Hello! I'm interested in working on this and have a couple of ideas on how this can be implemented. It doesn't seem as though there's been much progress made on this (given that the last bit of responses were made 3 years ago) so I would be happy to help out & work on developing this.

I've already proposed most of my ideas in a discussion I posted earlier
> At the moment, on Lazer, multiplayer lobbies don't show any difficulty information about queued beatmaps anywhere. While you can right click them and press on details to view the difficulty information on the web, this doesn't update to account for applied mods (such as dt).
> 
> Because of this, it would be wise to add this information under the tooltip that already exists (at the moment, it only shows star level though) when hovering over the circle. ![image](https://private-user-images.githubusercontent.com/69545310/292287621-505ef80b-a786-4867-88e5-da35d7eaa720.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxOTA3NjUsIm5iZiI6MTcwMzE5MDQ2NSwicGF0aCI6Ii82OTU0NTMxMC8yOTIyODc2MjEtNTA1ZWY4MGItYTc4Ni00ODY3LTg4ZTUtZGEzNWQ3ZWFhNzIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDIwMjc0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRmMWExNmU0NTRmY2JiZTdmNzEzNmY2Yjg1ODEyYjY0OWJkMzcwMGExNzBhNTcwMDUwMzk2NzczOThiZjUwOWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2QBbz_KDYRat6VOxjd-VkIB9-wuvCRM43h7XtwgE3oU) This would achieve the same effect as hovering over a beatmap in stable multiplayer lobbies and being able to see the difficulty stats there.
> 
> The information that should be present is the AR, HP, CS, OD, BPM, length, and # of objects. Could also add the pass % but I'm not sure if the client is able to access that.

I believe that the best approach is to just grab the difficulty data and just slap it into the tooltip that already exists.

**Comment by bdach** [comment]
The design is the primary blocker here. I think your link is broken, which doesn't help.

**Comment by smallketchup82** [comment]
> The design is the primary blocker here. 

I don't really see anywhere that the data could be placed outside of a tooltip. Most of the screen real estate is being used up by other stuff.

My main idea is to put the data into the toolbox to match stable. I think it would be good to have the BPM, length, and number of objects on one row, the AR, HP, CS, and the OD values on another row.
![image](https://github.com/ppy/osu/assets/69545310/16df8172-07e2-41e9-b7b4-152e79ef9b64)

But I guess another idea could be to maybe halve the height of the chatbox and put all of the information above the chatbox.
![image](https://github.com/ppy/osu/assets/69545310/18a45cce-ae8d-42bc-957a-edb9e91af593)

I'm not much of a UI designer but I'll try to roughly sketch out both of these ideas soon.
> I think your link is broken, which doesn't help.

Weird, heres an [imgur link](https://imgur.com/a/Z4YmH1i) instead, hopefully it works

**Comment by smallketchup82** [comment]
I've managed to program some concept designs in for the first idea (fitting them in the tooltip).
![image](https://github.com/ppy/osu/assets/69545310/0fe1cabc-5106-4bbe-bd09-c32d5493b009)

I've also got this iteration
![image](https://github.com/ppy/osu/assets/69545310/227a35f0-3e39-47f8-ab1c-d063c3929fa1)

Not sure which one looks better
Also I weirdly stumbled upon a bug while making this. The star rating counter that's already implemented within the tooltip doesn't update to reflect the new star rating with mods applied. And neither does the stats which I've implemented. Not sure where exactly to get the correct information after accounting for mods, so I'll put that off for now. I'll probably come back and fix all of that at a later stage.

**Comment by smallketchup82** [comment]
> Also I weirdly stumbled upon a bug while making this. The star rating counter that's already implemented within the tooltip doesn't update to reflect the new star rating with mods applied. And neither does the stats which I've implemented. Not sure where exactly to get the correct information after accounting for mods, so I'll put that off for now. I'll probably come back and fix all of that at a later stage.

> I think the arrangement of the stats might be off

Fixed both of these issues.

Nomod
![image](https://github.com/ppy/osu/assets/69545310/b15bcd00-93c2-454d-b5d6-0b3287e3e854)

With DT
![image](https://github.com/ppy/osu/assets/69545310/4127ac9d-c188-4e14-a265-e4596eba048f)

Still need to figure out a way to make the star rating change. It seems like that's going to be particularly hard, so I might need a bit of help with that. I'll continue to work on it in [my branch](https://github.com/smallketchup82/osu/tree/multiplayer-difficulty-tooltip).

**Comment by peppy** [comment]
I'd use the version without pipes. I think it might be okay until we have a better design.

**Comment by akiya-nagatsuka** [comment]
@smallketchup82 in the latest screenshots you have provided the star rating is still 6.24* even with DT applied though you said it's fixed. Can please check again this?

**Comment by smallketchup82** [comment]
@smallketchup82 in the latest screenshots you have provided the star rating is still 6.24* even with DT applied though you said it's fixed. Can please check again this?

Yeah I'm actively trying to fix this as I said in the last message. This is pretty hard though, so It might take me a bit of time. I'll probably need help with it

**Comment by smallketchup82** [comment]
I've spent a while already trying to find a way to get the star rating to update based on what mods are applied. As I detailed before, I figured that using `difficultyCache.GetDifficultyAsync()` would be the best way to go about it, so I tried to implement it. But it didn't work, no matter how I tried to implement it. It would always just return the nomod star rating irrespective of the mods applied (even though the list of mods were being passed to it, and I could confirm this with 100% certainty).

I'm honestly out of ideas for other ways to grab the adjusted star rating with mods applied, so I'm tempted to just PR what I've gotten so far and creating a new issue to address the static star rating.
